cpp/scripts/gitutils.py

```diff
@@ -0,0 +1,134 @@
+
+
+def __gitdiff(*opts):
+    """Runs a git diff command with no pager set"""
+    return __git("--no-pager", "diff", *opts)
+
+
+def branch():
+    """Returns the name of the current branch"""
+    name = __git("rev-parse", "--abbrev-ref", "HEAD")
+    name = name.rstrip()
+    return name
+
+
+def uncommittedFiles():
+    """
+    Returns a list of all changed files that are not yet committed. This
+    means both untracked/unstaged as well as uncommitted files too.
+    """
+    files = __gitdiff("--name-only", "--cached", "--ignore-submodules")
+    ret = files.splitlines()
+    files = __git("status", "-u", "-s")
+    for f in files.splitlines():
+        f = f.decode(encoding='UTF-8')
+        f = f.strip(" ")
+        tmp = f.split(" ", 1)
+        ret.append(tmp[1])
+    return ret
+
+
+def changedFilesBetween(b1, b2):
+    """Returns a list of files changed between branches b1 and b2"""
+    current = branch()
+    __git("checkout", "--quiet", b1)
+    __git("checkout", "--quiet", b2)
+    files = __gitdiff("--name-only", "--ignore-submodules", "%s...%s" %
+                      (b1, b2))
+    __git("checkout", "--quiet", current)
+    return files.splitlines()
+
+
+def changesInFileBetween(file, b1, b2, filter=None):
+    """Filters the changed lines to a file between the branches b1 and b2"""
+    current = branch()
+    __git("checkout", "--quiet", b1)
+    __git("checkout", "--quiet", b2)
+    diffs = __gitdiff("--ignore-submodules", "-w", "--minimal", "-U0",
+                      "%s...%s" % (b1, b2), "--", file)
+    __git("checkout", "--quiet", current)
+    lines = []
+    for line in diffs.splitlines():
+        if filter is None or filter(line):
+            lines.append(line)
+    return lines
+
+
+def modifiedFiles(filter=None):
+    """
+    If inside a CI-env (ie. currentBranch=current-pr-branch and the env-var
+    PR_TARGET_BRANCH is defined), then lists out all files modified between
+    these 2 branches. Else, lists out all the uncommitted files in the current
+    branch.
+
+    Such utility function is helpful while putting checker scripts as part of
+    cmake, as well as CI process. This way, during development, only the files
+    touched (but not yet committed) by devs can be checked. But, during the CI
+    process ALL files modified by the dev, as submiited in the PR, will be
+    checked. This happens, all the while using the same script.
+    """
+    if "PR_TARGET_BRANCH" in os.environ and branch() == "current-pr-branch":
+        allFiles = changedFilesBetween(os.environ["PR_TARGET_BRANCH"],
+                                       branch())
+    else:
+        allFiles = uncommittedFiles()
+    files = []
+    for f in allFiles:
+        if filter is None or filter(f):
+            files.append(f)
+    return files
+
+
+def listAllFilesInDir(folder):
+    """Utility functino to list all files/subdirs in the input folder"""
+    files = []
+    for f in os.listdir(folder):
```

Comment:
I think you could do `return [str(x) for x in pathlib.Path(folder).glob('**/*')]` to cover this function more concisely

Reply:
Ah.. my bad, in #618, @raydouglass suggested to use `os.walk` instead. Thus, I updated this method to use it, but forgot to update this PR as well. Let me merge the latest of branch-0.8 to make sure everything is up-to-date.

Reply:
Updated now. Thanks for catching this.